conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalytics.java

```diff
@@ -623,6 +631,41 @@ public void onSeeked(SeekedEvent seekedEvent) {
     };
     // endregion
 
+    // region timeshift events
+    private OnTimeShiftListener onTimeShiftListener = new OnTimeShiftListener() {
+        @Override
+        public void onTimeShift(TimeShiftEvent timeShiftEvent) {
+            if (!isSessionActive()) {
+                // See comment in onSeek
+                return;
+            }
+            Log.d(TAG, "[Player Event] OnTimeShift");
+            try {
+                // According to conviva it is valid to pass -1 for seeking in live streams
+                playerStateManager.setPlayerSeekStart(-1);
+            } catch (ConvivaException e) {
+                Log.e(TAG, e.getLocalizedMessage());
+            }
+        }
+    };
+
+    private OnTimeShiftedListener onTimeShiftedListener = new OnTimeShiftedListener() {
+        @Override
+        public void onTimeShifted(TimeShiftedEvent timeShiftedEvent) {
+            if (!isSessionActive()) {
+                // See comment in onSeek
+                return;
+            }
+            Log.d(TAG, "[Player Event] OnTimeShifted");
+            try {
+                playerStateManager.setPlayerSeekEnd();
+            } catch (ConvivaException e) {
+                Log.e(TAG, e.getLocalizedMessage());
+            }
+        }
```

Comment:
IMO this should be extracted in a separate method and called from `onTimeShiftedListener` and `onSeekedListener`.

Reply:
Done